diagnostics-app/src/app/core/services/routine-v2.service.ts

```diff
@@ -0,0 +1,237 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Service for starting to capture events and stopping
+ * capturing events from Chrome extension.
+ */
+
+import {Injectable} from '@angular/core';
+import {Subject} from 'rxjs';
+
+import {VISIBLE_ROUTINE_V2_CARDS} from 'common/config';
+import {
+  CancelRoutineMessage,
+  CreateRoutineMessage,
+  IsCreateRoutineResponse,
+  IsRoutineArgumentSupportedMessage,
+  IsRoutineSupportStatusInfo,
+  PortName,
+  Request,
+  RequestType,
+  Response,
+  ResponseErrorInfoMessage,
+  RoutineV2Action,
+  RoutineV2Argument,
+  RoutineV2Event,
+  RoutineV2Request,
+  RoutineV2Response,
+  StartRoutineMessage,
+} from 'common/message';
+import {
+  CreateRoutineResponse,
+  RoutineSupportStatus,
+  RoutineSupportStatusInfo,
+} from 'common/telemetry-extension-types';
+import {environment} from 'environments/environment';
+
+export interface getSubjectResponse {
+  success: Boolean;
+  subject?: Subject<RoutineV2Event>;
+  error?: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class RoutineV2Service {
+  // the ID of the extension service connects to.
+  private extensionId: string = environment.extensionId;
+  // A map storing subjects to forward the runtime responses from running routine V2 to.
+  private subjects = new Map<RoutineV2Argument, Subject<RoutineV2Event>>();
+  // the port for connecting with the extension to receive runtime responses.
+  private port?: chrome.runtime.Port;
+  // Since we use routine argument as the source of truth for each routine type,
```

Comment:
Should the first character in comment be capital? I found that the style is inconsistent.

Reply:
Yes. Let me make this style more consistent.

Reply:
I will create another PR that fixes all the comment related stuff altogether based on this change. For now, let's leave this as-is.

Reply:
Noted!